src/tests/ftest/aggregation/aggregation_throttling.py

```diff
@@ -0,0 +1,145 @@
+from ior_utils import IorCommand, IorMetrics
+
+class DaosAggregationThrottling(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description:
+
+       Verify ior performance during aggregation throttling
+
+    :avocado: recursive
+    """
+
+    def test_aggregation_throttling(self):
+        """Jira ID: DAOS-3749
+
+        Test Description:
+            Verify the ior throttling during aggregation
+            in the background is affecting the ior
+            performance only by +/- 10%
+        Use case:
+            Create a pool and container
+            Disable the aggregation
+            Run ior with same file option
+            Capture the initial ior performance.
+            Run ior the second time with same file option, so the
+            logical partitioning can be overwritten.
+            Enable the aggregation and wait for 90 seconds
+            Now as the aggregation is running in the background, run
+            ior again so both aggregation and ior runs in parallel
+            Capture the ior performance now and verify that it is
+            +/- 10% of the initial performance.
+            Also, verify the aggregation reclaimed the space used by
+            second ior.
+
+        :avocado: tags=all,pr,hw,large,aggregate,daosio,aggregatethrottling
+        """
+
+        # Create pool and container
+        self.update_ior_cmd_with_pool()
+
+        # Disable the aggregation
+        self.pool.set_property("reclaim", "disabled")
+
+        # Run ior with -k option to retain the file and not delete it
+        out = self.run_ior_with_pool()
+        metric_before_aggregate = IorCommand.get_ior_metrics(out)
+
+        # Run ior the second time on the same pool and container, so another
+        # copy of the file is inserted in DAOS.
+        out = self.run_ior_with_pool(update=False)
+
+        # wait 90 seconds for files to get old enough for aggregation
+        self.log.info("Waiting for 90 seconds for aggregation to start")
+        time.sleep(90)
+        # Enable the aggregation
+        self.pool.set_property("reclaim", "time")
+
+        # Run ior the third time while the aggregation of first two runs
+        # are running in the background.
+        out = self.run_ior_with_pool(update=False)
+        metric_after_aggregate = IorCommand.get_ior_metrics(out)
+
+        self.verify_performance(metric_before_aggregate,
+                                metric_after_aggregate,
+                                0, # write_perf
+                                10.0) # 10% perf difference
+
+        self.verify_performance(metric_before_aggregate,
+                                metric_after_aggregate,
+                                1, # read_perf
+                                10.0)
+
+    def verify_performance(self, before_metric, after_metric, read_write_idx,
+                           expected_perf_diff):
+        """Verify the after_metric read/write performances are within 
```

Comment:
(style)  trailing whitespace
(pylint-trailing-whitespace) Trailing whitespace

Reply:
Please fix the trailing white space issue.

Reply:
Done